sys/net/routing/rpl/rpl_control_messages.c

```diff
@@ -909,21 +906,28 @@ void rpl_recv_DIS(void)
 
                 rpl_opt_solicited_buf = get_rpl_opt_solicited_buf(len);
 
-                if (rpl_opt_solicited_buf->VID_Flags & RPL_DIS_I_MASK) {
-                    if (my_dodag->instance->id != rpl_opt_solicited_buf->rplinstanceid) {
-                        return;
-                    }
-                }
+                for (dodag = rpl_dodags, end = dodag + RPL_MAX_DODAGS; dodag < end; dodag++) {
+                    if (dodag->joined) {
+                        if (rpl_opt_solicited_buf->VID_Flags & RPL_DIS_I_MASK) {
+                            if (dodag->instance->id != rpl_opt_solicited_buf->rplinstanceid) {
+                                continue;
+                            }
+                        }
 
-                if (rpl_opt_solicited_buf->VID_Flags & RPL_DIS_D_MASK) {
-                    if (!rpl_equal_id(&my_dodag->dodag_id, &rpl_opt_solicited_buf->dodagid)) {
-                        return;
-                    }
-                }
+                        if (rpl_opt_solicited_buf->VID_Flags & RPL_DIS_D_MASK) {
+                            if (!rpl_equal_id(&dodag->dodag_id, &rpl_opt_solicited_buf->dodagid)) {
+                                continue;
+                            }
+                        }
+
+                        if (rpl_opt_solicited_buf->VID_Flags & RPL_DIS_V_MASK) {
+                            if (dodag->version != rpl_opt_solicited_buf->version) {
+                                continue;
+                            }
+                        }
 
-                if (rpl_opt_solicited_buf->VID_Flags & RPL_DIS_V_MASK) {
-                    if (my_dodag->version != rpl_opt_solicited_buf->version) {
-                        return;
+                        rpl_send_DIO(dodag, &ipv6_buf->srcaddr);
+                        trickle_reset_timer(&dodag->trickle);
```

Comment:
[RFC6550](https://tools.ietf.org/html/rfc6550#page-58)
>The Solicited Information option contains flags that indicate which
   predicates a node should check when deciding whether to reset its
   Trickle timer.  A node resets its Trickle timer when all predicates
   are true.  If a flag is set, then the RPL node MUST check the
   associated predicate.  If a flag is cleared, then the RPL node MUST
   NOT check the associated predicate.  (If a flag is cleared, the RPL
   node assumes that the associated predicate is true.)